common:
    lid_topic: "/livox/lidar"
    imu_topic: "/livox/imu"
    reloc_topic: "/slaver/robot_command"
    mode_topic: "/slaver/robot_status"
    time_sync_en: false # ONLY turn on when external time synchronization is really not possible
    time_offset_lidar_to_imu:
        0.001792 # Time offset between lidar and IMU calibrated by other algorithms, e.g. LI-Init (can be found in README).
        # This param will take effect no matter what time_sync_en is. So if the time offset is not known exactly, please set as 0.0

preprocess:
    lidar_type: 1 # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR, 3 for ouster LiDAR,
    scan_line: 4
    blind: 0.7
    point_filter_num: 1

mapping:
    acc_cov: 0.1
    gyr_cov: 0.1
    b_acc_cov: 0.0001
    b_gyr_cov: 0.0001
    fov_degree: 360.0
    det_range: 100.0
    extrinsic_est_en: false # true: enable the online estimation of IMU-LiDAR extrinsic
    extrinsic_T: [-0.011, -0.0234, 0.044] # DJI给的
    extrinsic_R: [1, 0, 0, 0, 1, 0, 0, 0, 1]

publish:
    path_en: true
    scan_publish_en: true # false: close all the point cloud output
    dense_publish_en: true # false: low down the points number in a global-frame point clouds scan.
    scan_bodyframe_pub_en: true # true: output the point cloud scans in IMU-body-frame

pcd_save:
    pcd_save_en: true
    interval:
        -1 # how many LiDAR frames saved in each pcd file;
        # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.

max_iteration: 3
filter_size_surf: 0.2
filter_size_map: 0.2
cube_side_length: 1000
runtime_pos_log_enable: 1
# init_method: PPICP # 建议使用PPICP ，NDT和ICP定位初始化非常容易崩溃
# NDT:
# ICP:
#     max_iter: 10
# PPICP:
#     plane_dist: 0.1
#     max_iter: 10

prior:
    red_prior_T: [0 ,0, 0]
    blue_prior_T: [0 ,0, 0]
    estimateGrav: true
    prior_R: [1 ,0, 0,
              0 ,1, 0,
              0 ,0, 1]
